File: practice_java_program/notpad/BothStringAngrambyHashmap.java
import java.util.*;
public class BothStringAngrambyHashmap{
	public static void main(String[] args){
		String input1 = "listen";
		String input2 = "silent";
		if(input1.length()!=input2.length()){
			System.out.println("Strings are not anagram");
			return;
		}
		HashMap<Character, Integer> hs = new HashMap<>();
		for(char ch:input1.toCharArray()){
			hs.put(ch, hs.getOrDefault(ch,0)+1);
		}
		for(char ch:input2.toCharArray()){
			if(!hs.containsKey(ch)){
				System.out.println("not anagram");
				return;
			}
			hs.put(ch, hs.get(ch)-1);
			if(hs.get(ch)==0){
				hs.remove(ch);
			}
		}
		if(hs.isEmpty()==true){
			System.out.println("anagram");
		}
		else{
			System.out.println("not anagram");
		}
	}
}

File: python_oriented_interview/notpad/BothStringAngrambyHashmap.py
def check_anagram():
    input1 = "listen"
    input2 = "silent"
    if len(input1) != len(input2):
        print("Strings are not anagram")
        return
    
    hs = {}
    for ch in input1:
        hs[ch] = hs.get(ch, 0) + 1
        
    for ch in input2:
        if ch not in hs:
            print("not anagram")
            return
        hs[ch] -= 1
        if hs[ch] == 0:
            del hs[ch]
            
    if len(hs) == 0:
        print("anagram")
    else:
        print("not anagram")

if __name__ == "__main__":
    check_anagram()
